From 7736a49df1ea94c2d5a27867076508b42a0d8b00 Mon Sep 17 00:00:00 2001
From: Kyle Repinski <repinski23@gmail.com>
Date: Thu, 19 Nov 2015 18:15:00 -0600
Subject: [PATCH 01/13] camera: Makefile overhaul.

Upon trying to build only the OMX interface, it was discovered that some
necessary bits and pieces were only in the ALL interface. After attempting
to fix this, it became painfully obvious that the entire makefile should
needed a lot of sprucing up and updating.

The USB-only interface does not build, but it didn't before this either.
There's a number of (D)OMX ties in the common files that prevent it.
Considering nobody has noticed this, it's probably not really used.
Should somebody find they need it, they can try and fix it themselves.

Change-Id: I5a99727fa37371172132da28b697d754b20005de
---
 camera/Android.mk | 193 +++++++++++++++++++++++++++---------------------------
 1 file changed, 96 insertions(+), 97 deletions(-)

diff --git a/camera/Android.mk b/camera/Android.mk
index 5f78a9d..bde06aa 100755
--- a/camera/Android.mk
+++ b/camera/Android.mk
@@ -1,70 +1,74 @@
 LOCAL_PATH:= $(call my-dir)
 
-#OMAP4_CAMERA_HAL_USES:= OMX
-#OMAP4_CAMERA_HAL_USES:= USB
-OMAP4_CAMERA_HAL_USES:= ALL
+TI_CAMERAHAL_INTERFACE ?= ALL
 
-CAMERAHAL_CFLAGS += $(ANDROID_API_CFLAGS)
+TI_CAMERAHAL_COMMON_CFLAGS := \
+    $(ANDROID_API_CFLAGS) \
+    -DLOG_TAG=\"CameraHal\" \
+    -DCOPY_IMAGE_BUFFER \
+    -fno-short-enums
 
 ifdef TI_CAMERAHAL_DEBUG_ENABLED
     # Enable CameraHAL debug logs
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_DEBUG
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_DEBUG
 endif
 
 ifdef TI_CAMERAHAL_VERBOSE_DEBUG_ENABLED
     # Enable CameraHAL verbose debug logs
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_DEBUG_VERBOSE
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_DEBUG_VERBOSE
 endif
 
 ifdef TI_CAMERAHAL_DEBUG_FUNCTION_NAMES
     # Enable CameraHAL function enter/exit logging
-    CAMERAHAL_CFLAGS += -DTI_UTILS_FUNCTION_LOGGER_ENABLE
+    TI_CAMERAHAL_COMMON_CFLAGS += -DTI_UTILS_FUNCTION_LOGGER_ENABLE
 endif
 
 ifdef TI_CAMERAHAL_DEBUG_TIMESTAMPS
     # Enable timestamp logging
-    CAMERAHAL_CFLAGS += -DTI_UTILS_DEBUG_USE_TIMESTAMPS
+    TI_CAMERAHAL_COMMON_CFLAGS += -DTI_UTILS_DEBUG_USE_TIMESTAMPS
 endif
 
 ifndef TI_CAMERAHAL_DONT_USE_RAW_IMAGE_SAVING
     # Enabled saving RAW images to file
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_USE_RAW_IMAGE_SAVING
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_USE_RAW_IMAGE_SAVING
 endif
 
 ifdef TI_CAMERAHAL_PROFILING
     # Enable OMX Camera component profiling
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_OMX_PROFILING
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_OMX_PROFILING
 endif
 
 ifdef TI_CAMERAHAL_MAX_CAMERAS_SUPPORTED
-    CAMERAHAL_CFLAGS += -DMAX_CAMERAS_SUPPORTED=$(TI_CAMERAHAL_MAX_CAMERAS_SUPPORTED)
+    TI_CAMERAHAL_COMMON_CFLAGS += -DMAX_CAMERAS_SUPPORTED=$(TI_CAMERAHAL_MAX_CAMERAS_SUPPORTED)
 endif
 
 ifdef TI_CAMERAHAL_TREAT_FRONT_AS_BACK
-    CAMERAHAL_CFLAGS += -DTREAT_FRONT_AS_BACK
+    TI_CAMERAHAL_COMMON_CFLAGS += -DTREAT_FRONT_AS_BACK
 endif
 
 ifeq ($(findstring omap5, $(TARGET_BOARD_PLATFORM)),omap5)
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_OMAP5_CAPTURE_MODES
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_OMAP5_CAPTURE_MODES
 endif
 
 ifeq ($(ENHANCED_DOMX),true)
-    CAMERAHAL_CFLAGS += -DENHANCED_DOMX
+    TI_CAMERAHAL_COMMON_CFLAGS += -DENHANCED_DOMX
 endif
 
 ifdef ARCH_ARM_HAVE_NEON
-    CAMERAHAL_CFLAGS += -DARCH_ARM_HAVE_NEON
+    TI_CAMERAHAL_COMMON_CFLAGS += -DARCH_ARM_HAVE_NEON
 endif
 
 ifeq ($(BOARD_VENDOR),motorola-omap4)
-    CAMERAHAL_CFLAGS += -DMOTOROLA_CAMERA
+    TI_CAMERAHAL_COMMON_CFLAGS += -DMOTOROLA_CAMERA
 endif
 
 ifeq ($(TARGET_BOOTLOADER_BOARD_NAME),piranha)
-    CAMERAHAL_CFLAGS += -DCAMERAHAL_PIRANHA
+    TI_CAMERAHAL_COMMON_CFLAGS += -DCAMERAHAL_PIRANHA
 endif
 
-CAMERAHAL_CFLAGS += -DLOG_TAG=\"CameraHal\"
+TI_CAMERAHAL_OMX_CFLAGS := -DOMX_CAMERA_ADAPTER
+TI_CAMERAHAL_USB_CFLAGS := -DV4L_CAMERA_ADAPTER
+
 
 TI_CAMERAHAL_COMMON_INCLUDES := \
     $(LOCAL_PATH)/../include \
@@ -72,7 +76,8 @@ TI_CAMERAHAL_COMMON_INCLUDES := \
     external/jpeg \
     external/jhead \
     $(LOCAL_PATH)/../libtiutils \
-    $(LOCAL_PATH)/inc
+    $(LOCAL_PATH)/inc \
+    system/media/camera/include
 
 ifdef ANDROID_API_JB_OR_LATER
 TI_CAMERAHAL_COMMON_INCLUDES += \
@@ -82,6 +87,16 @@ TI_CAMERAHAL_COMMON_INCLUDES += \
     frameworks/base/include/media/stagefright
 endif
 
+TI_CAMERAHAL_OMX_INCLUDES := \
+    frameworks/native/include/media/openmax \
+    $(DOMX_PATH)/mm_osal/inc \
+    $(DOMX_PATH)/omx_core/inc \
+    $(LOCAL_PATH)/inc/OMXCameraAdapter
+
+TI_CAMERAHAL_USB_INCLUDES := \
+    $(LOCAL_PATH)/inc/V4LCameraAdapter
+
+
 TI_CAMERAHAL_COMMON_SRC := \
     CameraHal_Module.cpp \
     CameraHal.cpp \
@@ -119,10 +134,29 @@ TI_CAMERAHAL_OMX_SRC := \
     OMXCameraAdapter/OMXZoom.cpp \
     OMXCameraAdapter/OMXDccDataSave.cpp
 
+ifdef TI_CAMERAHAL_USES_LEGACY_DOMX_DCC
+TI_CAMERAHAL_OMX_CFLAGS += -DUSES_LEGACY_DOMX_DCC
+else
+TI_CAMERAHAL_OMX_SRC += OMXCameraAdapter/OMXDCC.cpp
+endif
+
 TI_CAMERAHAL_USB_SRC := \
     V4LCameraAdapter/V4LCameraAdapter.cpp \
     V4LCameraAdapter/V4LCapabilities.cpp
 
+
+TI_CAMERAHAL_EXIF_LIBRARY := libexif
+# libexif is now libjhead in later API levels.
+
+ifdef ANDROID_API_KK_OR_LATER
+ifdef ANDROID_API_LP_OR_LATER
+    TI_CAMERAHAL_EXIF_LIBRARY := libjhead
+else ifneq ($(filter 4.4.3 4.4.4,$(PLATFORM_VERSION)),)
+    # Only 4.4.3 and 4.4.4 KK use libjhead
+    TI_CAMERAHAL_EXIF_LIBRARY := libjhead
+endif
+endif
+
 TI_CAMERAHAL_COMMON_SHARED_LIBRARIES := \
     libui \
     libbinder \
@@ -132,158 +166,123 @@ TI_CAMERAHAL_COMMON_SHARED_LIBRARIES := \
     libtiutils \
     libcamera_client \
     libgui \
-    libjpeg
-
-ifeq ($(shell test $(PLATFORM_SDK_VERSION) -ge 19 || echo 1),)
-# check for 5.0 and greater
-ifeq ($(shell test $(PLATFORM_SDK_VERSION) -ge 21 || echo 1),)
-TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
-    libjhead
-# currently 4.4.3 and 4.4.4
-else ifneq ($(filter 4.4.3 4.4.4,$(PLATFORM_VERSION)),)
-TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
-    libjhead
-else
-TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
-    libexif
-endif
-else
-TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
-    libexif
-endif
+    libjpeg \
+    $(TI_CAMERAHAL_EXIF_LIBRARY)
 
 ifdef ANDROID_API_JB_MR1_OR_LATER
 TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
     libion_ti
+TI_CAMERAHAL_COMMON_CFLAGS += -DUSE_LIBION_TI
 else
 TI_CAMERAHAL_COMMON_SHARED_LIBRARIES += \
     libion
 endif
 
+TI_CAMERAHAL_OMX_SHARED_LIBRARIES := \
+    libmm_osal \
+    libOMX_Core \
+    libdomx
+
+
 ifdef OMAP_ENHANCEMENT_CPCAM
 TI_CAMERAHAL_COMMON_STATIC_LIBRARIES += \
     libcpcamcamera_client
 endif
 
 
+ifeq ($(TI_CAMERAHAL_INTERFACE),OMX)
 # ====================
 #  OMX Camera Adapter
 # --------------------
 
-ifeq ($(OMAP4_CAMERA_HAL_USES),OMX)
-
 include $(CLEAR_VARS)
 
-CAMERAHAL_CFLAGS += -DOMX_CAMERA_ADAPTER
-
-LOCAL_SRC_FILES:= \
+LOCAL_SRC_FILES := \
     $(TI_CAMERAHAL_COMMON_SRC) \
     $(TI_CAMERAHAL_OMX_SRC)
 
-LOCAL_C_INCLUDES += \
+LOCAL_C_INCLUDES := \
     $(TI_CAMERAHAL_COMMON_INCLUDES) \
-    $(DOMX_PATH)/omx_core/inc \
-    $(DOMX_PATH)/mm_osal/inc \
-    frameworks/native/include/media/openmax \
-    $(LOCAL_PATH)/inc/OMXCameraAdapter
+    $(TI_CAMERAHAL_OMX_INCLUDES)
 
-LOCAL_SHARED_LIBRARIES:= \
+LOCAL_SHARED_LIBRARIES := \
     $(TI_CAMERAHAL_COMMON_SHARED_LIBRARIES) \
-    libmm_osal \
-    libOMX_Core \
-    libdomx
+    $(TI_CAMERAHAL_OMX_SHARED_LIBRARIES)
 
 LOCAL_STATIC_LIBRARIES := $(TI_CAMERAHAL_COMMON_STATIC_LIBRARIES)
 
-LOCAL_CFLAGS := -fno-short-enums -DCOPY_IMAGE_BUFFER $(CAMERAHAL_CFLAGS)
+LOCAL_CFLAGS := \
+    $(TI_CAMERAHAL_COMMON_CFLAGS) \
+    $(TI_CAMERAHAL_OMX_CFLAGS)
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
-LOCAL_MODULE:= camera.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_TAGS:= optional
+LOCAL_MODULE := camera.$(TARGET_BOARD_PLATFORM)
+LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_HEAPTRACKED_SHARED_LIBRARY)
 
-else
-ifeq ($(OMAP4_CAMERA_HAL_USES),USB)
-
-
+else ifeq ($(TI_CAMERAHAL_INTERFACE),USB)
 # ====================
 #  USB Camera Adapter
 # --------------------
 
 include $(CLEAR_VARS)
 
-CAMERAHAL_CFLAGS += -DV4L_CAMERA_ADAPTER
-
-LOCAL_SRC_FILES:= \
+LOCAL_SRC_FILES := \
     $(TI_CAMERAHAL_COMMON_SRC) \
     $(TI_CAMERAHAL_USB_SRC)
 
-LOCAL_C_INCLUDES += \
+LOCAL_C_INCLUDES := \
     $(TI_CAMERAHAL_COMMON_INCLUDES) \
-    $(LOCAL_PATH)/inc/V4LCameraAdapter
+    $(TI_CAMERAHAL_USB_INCLUDES)
 
-LOCAL_SHARED_LIBRARIES:= \
+LOCAL_SHARED_LIBRARIES := \
     $(TI_CAMERAHAL_COMMON_SHARED_LIBRARIES)
 
 LOCAL_STATIC_LIBRARIES := $(TI_CAMERAHAL_COMMON_STATIC_LIBRARIES)
 
-LOCAL_CFLAGS := -fno-short-enums -DCOPY_IMAGE_BUFFER $(CAMERAHAL_CFLAGS)
+LOCAL_CFLAGS := \
+    $(TI_CAMERAHAL_COMMON_CFLAGS) \
+    $(TI_CAMERAHAL_USB_CFLAGS)
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
-LOCAL_MODULE:= camera.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_TAGS:= optional
+LOCAL_MODULE := camera.$(TARGET_BOARD_PLATFORM)
+LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_HEAPTRACKED_SHARED_LIBRARY)
 
-else
-ifeq ($(OMAP4_CAMERA_HAL_USES),ALL)
-
-
+else ifeq ($(TI_CAMERAHAL_INTERFACE),ALL)
 # =====================
 #  ALL Camera Adapters
 # ---------------------
 
 include $(CLEAR_VARS)
 
-CAMERAHAL_CFLAGS += -DOMX_CAMERA_ADAPTER -DV4L_CAMERA_ADAPTER -DUSE_LIBION_TI
-
-LOCAL_SRC_FILES:= \
+LOCAL_SRC_FILES := \
     $(TI_CAMERAHAL_COMMON_SRC) \
     $(TI_CAMERAHAL_OMX_SRC) \
     $(TI_CAMERAHAL_USB_SRC)
 
-LOCAL_C_INCLUDES += \
+LOCAL_C_INCLUDES := \
     $(TI_CAMERAHAL_COMMON_INCLUDES) \
-    $(DOMX_PATH)/omx_core/inc \
-    $(DOMX_PATH)/mm_osal/inc \
-    frameworks/native/include/media/openmax \
-    $(LOCAL_PATH)/inc/OMXCameraAdapter \
-    $(LOCAL_PATH)/inc/V4LCameraAdapter \
-    system/media/camera/include
+    $(TI_CAMERAHAL_OMX_INCLUDES) \
+    $(TI_CAMERAHAL_USB_INCLUDES)
 
-LOCAL_SHARED_LIBRARIES:= \
+LOCAL_SHARED_LIBRARIES := \
     $(TI_CAMERAHAL_COMMON_SHARED_LIBRARIES) \
-    libmm_osal \
-    libOMX_Core \
-    libdomx
+    $(TI_CAMERAHAL_OMX_SHARED_LIBRARIES)
 
 LOCAL_STATIC_LIBRARIES := $(TI_CAMERAHAL_COMMON_STATIC_LIBRARIES)
 
-LOCAL_CFLAGS := -fno-short-enums -DCOPY_IMAGE_BUFFER $(CAMERAHAL_CFLAGS)
-
-ifdef TI_CAMERAHAL_USES_LEGACY_DOMX_DCC
-LOCAL_CFLAGS += -DUSES_LEGACY_DOMX_DCC
-else
-LOCAL_SRC_FILES += OMXCameraAdapter/OMXDCC.cpp
-endif
+LOCAL_CFLAGS := \
+    $(TI_CAMERAHAL_COMMON_CFLAGS) \
+    $(TI_CAMERAHAL_OMX_CFLAGS) \
+    $(TI_CAMERAHAL_USB_CFLAGS)
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
-LOCAL_MODULE:= camera.$(TARGET_BOARD_PLATFORM)
-LOCAL_MODULE_TAGS:= optional
+LOCAL_MODULE := camera.$(TARGET_BOARD_PLATFORM)
+LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_HEAPTRACKED_SHARED_LIBRARY)
 
 endif
-endif
-endif
-- 
2.7.4

